e2e/node/e2e.test.ts

```diff
@@ -33,156 +33,124 @@ import {
   // eslint-disable-next-line no-shadow
   beforeEach,
 } from "@jest/globals";
+
 import { Session } from "@inrupt/solid-client-authn-node";
-import { config } from "dotenv-flow";
-import * as openidClient from "openid-client";
 import {
-  getPodUrlAll,
   createContainerInContainer,
   deleteContainer,
   getSourceIri,
 } from "@inrupt/solid-client";
-import { WebsocketNotification } from "../../src/index";
-
-import { getTestingEnvironmentNode } from "../e2e-setup";
+import { WebsocketNotification, ErrorEvent } from "../../src/index";
 
-config({
-  path: __dirname,
-  // In CI, actual environment variables will overwrite values from .env files.
-  // We don't need warning messages in the logs for that:
-  silent: process.env.CI === "true",
-});
+import { getTestingEnvironmentNode } from "../utils/getTestingEnvironment";
+import {
+  getAuthenticatedSession,
+  getPodRoot,
+  setupTestResources,
+  teardownTestResources,
+} from "./test-helpers";
 
-const {
-  idp: oidcIssuer,
-  environment,
-  clientId,
-  clientSecret,
-  notificationGateway,
-  protocol,
-} = getTestingEnvironmentNode();
+const env = getTestingEnvironmentNode();
 
-const TEST_SLUG = "solid-client-test-e2e-notifications";
+const TEST_SLUG = "solid-client-notifications-test-e2e-resource";
 
-// Allows us to skip a test pending some conditions.
-const testIf = (condition: boolean) => (condition ? it : it.skip);
+const nextWebsocketMessage = async (ws: WebsocketNotification) => {
+  return new Promise((resolve) => {
+    // TODO: implement ws.once
```

Comment:
do we need this comment still?

Reply:
Yeah, we've a ticket on the backlog for it, as #246 starts to implement it; really we need this to clean up receiving messages for expectations, the way we're doing it now is very hacky/problematic.